chia/full_node/coin_store.py

```diff
@@ -435,12 +436,13 @@ async def rollback_to_block(self, block_index: int) -> List[CoinRecord]:
                     record = CoinRecord(coin, uint32(0), row[1], row[2], uint64(0))
                     coin_changes[record.name] = record
 
-            # Delete from storage
+            # Delete reverted blocks from storage
             await conn.execute("DELETE FROM coin_record WHERE confirmed_index>?", (block_index,))
 
+            # Add coins that are confirmed in the reverted blocks to the list of changed coins.
             async with conn.execute(
                 "SELECT confirmed_index, spent_index, coinbase, puzzle_hash, "
-                "coin_parent, amount, timestamp FROM coin_record WHERE confirmed_index>?",
+                "coin_parent, amount, timestamp FROM coin_record WHERE spent_index>?",
```

Comment:
this is the main bugfix, right? previously we would only consider coins that appeared after the fork-point as "changed", and now we also include coins that were spent after the fork point.

However, the fact that we delete all the new coins *first* would suggest that we're still failing to report those as changed.

Reply:
The reverted coins are being reported in line 429, I think. If a coin is both created and spent after the fork, it will only be reported as non-existing, not as unspent.